### PR TITLE
Convert language-specific option types to records

### PR DESCRIPTION
**Pull request by tmat**
We have a few option types that have a common base class and then C# and VB specific derived classes containing the language-specific settings. Ideally, all these types would be records. However, it is not currently possible to derive from C# record in VB.

We also eventually want some of these types to be public and thus have a good public surface and usability (e.g. being able to use `with` operator to get an updated options object).

To work around the VB limitation this change defines the VB type in common layer, in the same namespace it would have been defined in the VB layer if VB allowed the definition to exist there.

This wart seems like a reasonable trade-off, considering the alternative would be to hand-write (or source-generate) a lot of `WithXyz` methods, GetHashCode, Equals, etc.

Each commit can be reviewed separately.

### CODE REVIEW

File: src/Workspaces/VisualBasic/Portable/CodeCleanup/Providers/NormalizeModifiersOrOperatorsCodeCleanupProvider.vb [hunk 1]
```diff
@@ -52,7 +52,13 @@ Namespace Microsoft.CodeAnalysis.CodeCleanup.Providers
             ' this order will be used when the rewriter re-order modifiers
             ' PERF: Using UShort instead of SyntaxKind as the element type so that the compiler can use array literal initialization
             Private Shared ReadOnly s_modifierKindsInOrder As SyntaxKind() =
```
Comment (tmat) [inline src/Workspaces/VisualBasic/Portable/CodeCleanup/Providers/NormalizeModifiersOrOperatorsCodeCleanupProvider.vb:54]: Should this code use the current value from the options rather then defaults?

Comment (tmat) [reply src/Workspaces/VisualBasic/Portable/CodeCleanup/Providers/NormalizeModifiersOrOperatorsCodeCleanupProvider.vb:54]: If this is indeed correct, I'll add test that the order is the same as in VisualBasicIdeCodeStyleOptions.Default.

Comment (tmat) [reply src/Workspaces/VisualBasic/Portable/CodeCleanup/Providers/NormalizeModifiersOrOperatorsCodeCleanupProvider.vb:55]: Added assert.

**Review by dibarbet — APPROVED** [review]
seems reasonable

**Pull request merged**